packages/scripts/config/webpack.config.js

```diff
@@ -265,6 +260,21 @@ if ( ! isProduction ) {
 		use: require.resolve( 'source-map-loader' ),
 		enforce: 'pre',
 	} );
+	config.devServer = {
+		devMiddleware: {
+			writeToDisk: true,
+		},
+		allowedHosts: 'auto',
+		host: 'localhost',
+		port: 8887,
+		proxy: {
+			'/build': {
+				pathRewrite: {
+					'^/build': '',
+				},
+			},
+		},
```

Comment:
This is the best I could get so far. It automatically works with `ReactRefreshWebpackPlugin` for both Safari and Chrome. The only challenge would be when someone wants to use the port 8887 for something else. I choosed a port number similar to what `@wordpress/wp-env` uses by default: 8888 and 8889.